### PR TITLE
Recipe Suggestion Section

### LINKED ISSUE #328: FEATURE: Recipe For The Day Suggestion section
### Is there an existing issue raised for this?

- [X] I have searched the existing issues

### Describe the feature

Random Recipe: A feature that suggests a random recipe for users looking for inspiration.

### Component

I would like to add this feature on Homepage.

### It should be implemented because

Enhanced User Engagement: By showcasing a new, random recipe daily, users are encouraged to explore recipes they might not otherwise discover, increasing the time they spend on the platform.

Fresh and Dynamic Content: The homepage remains dynamic and interesting with new content appearing daily, making users more likely to return frequently to see what's new.

Recipe Discovery: It promotes lesser-known or newly contributed recipes, giving them more visibility and helping users broaden their culinary horizons.

User Inspiration: For users unsure of what to cook, the "Recipe of the Day" serves as a quick source of inspiration, solving decision fatigue.

### Would you like to work on this issue?

Yes

### Terms & Conditions

- [X] I agree to follow the Code of Conduct
- [X] I'm a GSSOC'24 Ext contributor

### PR DESCRIPTION
**Pull request by aditiverma-21**
## What does this PR do?
Adds a section for daily recipe suggestions

Fixes #328

### CODE REVIEW

**Comment by aditiverma-21** [comment]
@AlfiyaSiddique Please revoew this PR

File: backend/routes/recipes.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  try {
+    const count = await Recipe.countDocuments();
+    const randomIndex = Math.floor(Math.random() * count);
+    const randomRecipe = await Recipe.findOne().skip(randomIndex);
+    res.json(randomRecipe);
+  } catch (error) {
+    res.status(500).json({ message: 'Error fetching random recipe' });
+  }
+});
+
```
Comment (AlfiyaSiddique) [inline backend/routes/recipes.js:17]: @aditiverma-21 no need to create the route in this file there is a file with all routes named web,js
Please follow the code format

Comment (aditiverma-21) [reply backend/routes/recipes.js:17]: Okay sure I will remove this

**Comment by aditiverma-21** [comment]
@AlfiyaSiddique  Please check and let me know

File: frontend/src/Components/RecipeOfTheDay.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import React, { useEffect, useState } from 'react';
+
+function RecipeOfTheDay() {
+  const [recipe, setRecipe] = useState(null);
+
+  useEffect(() => {
+    // Fetch a random recipe from the API
+    fetch('/api/recipes/random')
+      .then((response) => response.json())
+      .then((data) => setRecipe(data))
```
Comment (AlfiyaSiddique) [inline frontend/src/Components/RecipeOfTheDay.jsx:11]: @aditiverma-21 you have used the random api here but where did you create this API??

**Pull request closed without merging**